### PR TITLE
Hydrates patient contacts not just parent contacts

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

Fixes what looks like a typo where the wrong array is hydrated.

medic/medic-webapp#4410

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

File: sentinel/test/unit/lineage.js [hunk 1]
```diff
@@ -434,6 +434,48 @@ exports['fetchHydratedDoc only fetches contacts that it has not got via lineage'
...
+  sinon.stub(db.medic, 'fetch').callsArgWith(1, null, { rows: [
+    { doc: dhContact },
+    { doc: report_parentContact }
+  ]});
+  sinon.stub(utils, 'getPatientContactUuid').callsArgWith(2, null, report_patient._id);
+
+  lineage.fetchHydratedDoc(docId).then(actual => {
+    test.equal(actual.patient.parent.parent.contact.phone, '+123');
+    test.done();
+  }).catch(test.done);
```
Comment (vimemo) [inline sentinel/test/unit/lineage.js]: we do have another test with this pattern but it is not completely clear to me what the intention is for test.done in both then & catch.

Comment (garethbowen) [reply sentinel/test/unit/lineage.js]: With nodeunit you need to call `test.done` for both failures and successes. This way the `err` gets passed to the test.done which fails the test with the err as the failure message. If you don't have this then nodeunit hangs waiting for the `test.done` to be called.

We're in the process of migrating to mocha which allows you to return the promise so this code would become...

```js
  return lineage.fetchHydratedDoc(docId).then(actual => {
    test.equal(actual.patient.parent.parent.contact.phone, '+123');
  });
```

File: sentinel/lib/lineage.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ const mergeLineages = (lineage, patientLineage) => {
   const lineageContacts = [],
         contactsToFetch = [];
   contactIds.forEach(id => {
-    const contact = lineage.find(d => d && d._id === id);
+    const contact = lineages.find(d => d && d._id === id);
```
Comment (SCdF) [inline sentinel/lib/lineage.js]: \*cough\* types \*cough\*

:shipit:

Comment (garethbowen) [reply sentinel/lib/lineage.js]: `lineage` and `lineages` have the same type... both arrays. I think "*cough* naming *cough*" is valid though.

**Pull request merged**